### PR TITLE
Improve PCA process performance

### PR DESCRIPTION
**Pull request by mzganec**

### CODE REVIEW

File: resolwe_bio/processes/clustering/pca.yml [hunk 1]
```diff
@@ -45,38 +45,27 @@
...
-      re-save proc.rc $?
+        --sample-files {% for exp in exps %} {{ exp.exp.file }} {% endfor %} \
+        --sample-ids {% for exp in exps %} {{ exp|sample_id }} {% endfor %} \
+        {% if genes %} --gene-labels {% for gene in genes %} {{ gene }} {% endfor %} {% endif %} \
+        {% if filter %} --skip-low-expressions {% endif %} \
+        --output-fn pca.json
+      re-checkrc "Principal component analysis failed."
+
+      re-save pca pca.json
+      re-progress 1.0
```
Comment (mstajdohar) [inline resolwe_bio/processes/clustering/pca.yml]: Not needed. Progress is set to 1 automatically when it's done.

File: resolwe_bio/processes/clustering/pca.yml [hunk 2]
```diff
@@ -45,38 +45,27 @@
...
-        {% else %}
-          {% if e.source != exps.0.source %}
-            re-error "All expressions must have the same source."
-          {% endif %}
-        {% endif %}
-      {% endfor %}
-
-      sample_files=()
-      sample_ids=()
+      {% if exps|selectattr('source', 'equalto', (exps|first).source)|list|length != exps|length %}
```
Comment (mstajdohar) [inline resolwe_bio/processes/clustering/pca.yml]: Why did you refactor this?

**Comment by mstajdohar** [comment]
@mzganec Please rebase

File: resolwe_bio/processes/clustering/pca.yml (hunk 2 above)
Comment (mzganec) [reply resolwe_bio/processes/clustering/pca.yml]: No particular reason. Can be changed back if `for` loops are preferred over one-liners.

**Pull request merged**